### PR TITLE
Fixed issue deadline not showing on issue page

### LINKED ISSUE #3922: Issue 'due date' format checking / 500 ISE
- Gitea version (or commit ref): 1.1.0+1205-gbb801488
- Git version:
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

The issue due date entry box is finicky:
- entering '1234' gives a HTTP 500 error page instead of an 'invalid format' warning
- entering '1234-01-01' makes parts of Gitea think a due date was entered, but others ignore it

## Screenshots

due-date-500

due-date-unset

### PR DESCRIPTION
**Pull request by kolaente**
Partially fixes #3922.

### CODE REVIEW

**Comment by daviian** [comment]
Should we generally prohibit due dates in the past? What do you say?

**Comment by kolaente** [comment]
@daviian Speaking from my experience, no. I admit probably noone will set a duedate prior to 1970, but at my company we sometimes have to set deadlines to like yesterday or one wwek ago.

**Comment by daviian** [comment]
@kolaente Ok if thats the case then guess the fix is fine for me 👍

**Review by lafriks — APPROVED** [review]
Tricky but I don't know anything better to recommend :)

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 1]
```diff
@@ -229,12 +229,12 @@
 		{{end}}
 
 		{{if and .IsSigned .IsRepositoryWriter}}
-			<form method="POST"{{if gt .Issue.DeadlineUnix 0}}style="display: none;"{{end}}} id="add_deadline_form" action="{{$.RepoLink}}/issues/{{.Issue.Index}}/deadline/update" class="ui action input fluid">
+			<form method="POST"{{if ne.Issue.DeadlineUnix 0}}style="display: none;"{{end}}} id="add_deadline_form" action="{{$.RepoLink}}/issues/{{.Issue.Index}}/deadline/update" class="ui action input fluid">
```
Comment (daviian) [inline templates/repo/issue/view_content/sidebar.tmpl]: missing space ;)

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 2]
```diff
@@ -229,12 +229,12 @@
...
 				{{$.CsrfTokenHtml}}
 				<div class="ui fluid action input">
-					<input required placeholder="{{.i18n.Tr "repo.issues.due_date_form"}}" {{if gt .Issue.DeadlineUnix 0}}value="{{.Issue.DeadlineUnix.Format "2006-01-02"}}"{{end}} type="date" name="date" style="min-width: 13.9rem;border-radius: 4px 0 0 4px;border-right: 0;white-space: nowrap;">
+					<input required placeholder="{{.i18n.Tr "repo.issues.due_date_form"}}" {{if ne.Issue.DeadlineUnix 0}}value="{{.Issue.DeadlineUnix.Format "2006-01-02"}}"{{end}} type="date" name="date" style="min-width: 13.9rem;border-radius: 4px 0 0 4px;border-right: 0;white-space: nowrap;">
```
Comment (daviian) [inline templates/repo/issue/view_content/sidebar.tmpl]: here as well

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 3]
```diff
@@ -229,12 +229,12 @@
...
 					<button class="ui green icon button">
-					{{if gt .Issue.DeadlineUnix 0}}
+					{{if ne.Issue.DeadlineUnix 0}}
```
Comment (daviian) [inline templates/repo/issue/view_content/sidebar.tmpl]: here as well

**Review by daviian — CHANGES_REQUESTED** [review]

**Review by daviian — APPROVED** [review]

**Pull request merged**